cmd/context/create.go

```diff
@@ -160,26 +160,45 @@ func (c *ContextCommand) UseContext(ctx context.Context, ctxOptions *ContextOpti
 			return err
 		}
 	}
-	if ctxOptions.IsOkteto && ctxOptions.Save {
-		okClient, err := c.OktetoClientProvider.Provide()
-		if err != nil {
-			return err
-		}
-		hasAccess, err := utils.HasAccessToNamespace(ctx, ctxOptions.Namespace, okClient)
-		if err != nil {
-			return err
+
+	if ctxOptions.Save {
+		hasAccess := false
+		if ctxOptions.IsOkteto {
+			okClient, err := c.OktetoClientProvider.Provide()
+			if err != nil {
+				return err
+			}
+			hasOktetoClientAccess, err := utils.HasAccessToOktetoClusterNamespace(ctx, ctxOptions.Namespace, okClient)
+			if err != nil {
+				return err
+			}
+
+			hasAccess = hasOktetoClientAccess
+		} else {
+			k8sClient, _, err := c.K8sClientProvider.Provide(okteto.Context().Cfg)
+			if err != nil {
+				return err
+			}
+
+			hasK8sClientAccess, err := utils.HasAccessToNamespace(ctx, ctxOptions.Namespace, k8sClient)
+			if err != nil {
+				return err
+			}
+
+			hasAccess = hasK8sClientAccess
 		}
```

Comment:
Could we extract this to an auxiliary function?

I also though about creating a struct for this, but it would have a lot of dependencies. WDYT?

Reply:
It seems like it deserves some testing which may mean that it needs to be pulled out into a function. 

It looks like we should be unit testing for...

- either an okteto client or a k8s client is created depending on the conditions
- correct access check is done.

Both of these things seem to be done as side-effect and therefore may be hard to test without some sort of re-factor.

Reply:
Also, after looking at the context a bit more the function is already >100 lines, which is a problem. I'd like to not make it worse. I think pulling it out and writing tests on it will be a good path for improving this bit of code.

Reply:
yes, totally agree. I will refactor it.